ethcore/src/miner/miner.rs

```diff
@@ -614,12 +615,19 @@ impl Miner {
 							}
 						}).unwrap_or(default_origin);
 
+						// try to install zgp-checker before appending transactions
```

Comment:
I think it's better to make sure you're using correct one, otherwise you can try to do some attacks on the syncing node (like feeding a chain with different ZGP contract, that will be re-organized later, but the ZGP contract stays the same)